### PR TITLE
Add Mac OS X support

### PR DESCRIPTION
**Pull request by jdek**
Adds support for building mpv-android on OS X, change the os var in path.sh to macosx to change os setting.

### CODE REVIEW

File: scripts/lua.sh [hunk 1]
```diff
@@ -13,7 +13,17 @@ fi
 
 # LUA_T= and LUAC_T= disable building lua & luac
 # -Dgetlocaledecpoint()=('.') fixes bionic missing decimal_point in localeconv
-make CC="arm-linux-androideabi-gcc -Dgetlocaledecpoint\(\)=\(\'.\'\)" PLAT=linux LUA_T= LUAC_T= -j6
+if [ "$os" == "macosx" ]; then
+	if hash ginstall 2>/dev/null; then
+		sed -i.bak "s/INSTALL= install/INSTALL= ginstall/" Makefile
```
Comment (sfan5) [inline scripts/lua.sh]: We'd like to avoid patching the Makefile, can't you pass this on the command line?

File: path.sh [hunk 2]
```diff
@@ -2,5 +2,13 @@
...
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-export PATH="$DIR/sdk/ndk-toolchain/bin:$DIR/sdk/android-ndk-r11c:$DIR/sdk/android-sdk-linux/tools:$PATH"
-export ANDROID_HOME="$DIR/sdk/android-sdk-linux"
+os=
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    os=linux
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    os=macosx
+fi
```
Comment (sfan5) [inline path.sh]:
``` bash
os=linux
[[ "$OSTYPE" == "darwin"* ]] && os=macosx
```

...would be a little shorter

File: scripts/lua.sh [hunk 3]
```diff
@@ -13,17 +13,27 @@ fi
...
+make CC="arm-linux-androideabi-gcc -Dgetlocaledecpoint\(\)=\(\'.\'\)" \
+     AR="arm-linux-androideabi-ar r" \
+     RANLIB="arm-linux-androideabi-ranlib" \
+     PLAT=linux LUA_T= LUAC_T= -j6
+
+if [ "$os" == "macosx" ]; then
+    INSTALL="ginstall"
+else
+    INSTALL="install"
+fi
```
Comment (sfan5) [inline scripts/lua.sh]:
``` bash
INSTALL=install
[ "$os" == "macosx" ] && INSTALL=ginstall
```

ditto

File: scripts/lua.sh [hunk 4]
```diff
@@ -13,17 +13,27 @@ fi
 
 # LUA_T= and LUAC_T= disable building lua & luac
 # -Dgetlocaledecpoint()=('.') fixes bionic missing decimal_point in localeconv
-make CC="arm-linux-androideabi-gcc -Dgetlocaledecpoint\(\)=\(\'.\'\)" PLAT=linux LUA_T= LUAC_T= -j6
+
```
Comment (sfan5) [inline scripts/lua.sh]: the comment belongs to the command below so it'd make sense not having an empty line inbetween imo

File: download.sh [hunk 5]
```diff
@@ -5,20 +5,37 @@ v_ndk=r11c
...
 
+. ./path.sh # load $os var
+
+if [ "$os" == "linux" ]; then
+	sdk_ext="tgz"
+	os_ndk="linux"
+elif [ "$os" == "macosx" ]; then
+	sdk_ext="zip"
+	os_ndk="darwin"
+    hash ginstall 2>/dev/null || { echo "Error: ginstall not found. You probably need GNU coreutils."; exit 255 }
```
Comment (sfan5) [inline download.sh]: s/spaces/tabs/

**Comment by jdek** [comment]
deps needed (note to self):
- automake
- autoconf
- libtool
- coreutils

Remember to document somewhere

**Pull request merged**